docs/fsharp-cheatsheet.md

```diff
@@ -169,197 +205,281 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
         if x = 0 then false
         else even (x - 1)
 
-<div id="pattern-matching"></div>
-
-## Pattern Matching
+<div id="statically-resolved-type-parameters"></div>
 
-Pattern matching is often facilitated through `match` keyword.
+### Statically Resolved Type Parameters
 
-    let rec fib n =
-        match n with
-        | 0 -> 0
-        | 1 -> 1
-        | _ -> fib (n - 1) + fib (n - 2)
+A *statically resolved type parameter* is a type parameter that is replaced with an actual type at compile time instead of at run time. They are primarily useful in conjunction with member constraints.
 
-In order to match sophisticated inputs, one can use `when` to create filters or guards on patterns:
+    let inline add x y = x + y
+    let integerAdd = add 1 2
+    let floatAdd = add 1.0f 2.0f // without `inline` on `add` function, this would cause a type error
 
-    let sign x =
-        match x with
-        | 0 -> 0
-        | x when x < 0 -> -1
-        | x -> 1
+####
 
-Pattern matching can be done directly on arguments:
+    type RequestA = { Id: string; StringValue: string }
+    type RequestB = { Id: string; IntValue: int }
 
-    let fst' (x, _) = x
+    let requestA: RequestA = { Id = "A"; StringValue = "Value" }
+    let requestB: RequestB = { Id = "B"; IntValue = 42 }
 
-or implicitly via `function` keyword:
+    let inline getId<'T when 'T : (member Id: string)> (x: 'T) = x.Id
 
-    /// Similar to `fib`; using `function` for pattern matching
-    let rec fib' = function
-        | 0 -> 0
-        | 1 -> 1
-        | n -> fib' (n - 1) + fib' (n - 2)
+    let idA = getId requestA  // "A"
+    let idB = getId requestB  // "B"
 
-For more complete reference visit [Pattern Matching (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/pattern-matching).
+See [Statically Resolved Type Parameters (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/generics/statically-resolved-type-parameters) and [Constraints (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/generics/constraints) for more examples.
 
 <div id="collections"></div>
 
 ## Collections
 
+<div id="collections-lists"></div>
+
 ### Lists
 
 A *list* is an immutable collection of elements of the same type.
```

Comment:
I guess its bad news to say cons array or linked list, but I do wish that point was conveyed relatively early on in people's experience